core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -142,8 +143,9 @@ private ServerBootstrap createBootstrap(EventLoopGroup group) {
                 if (System.getProperties().contains("disableNativeEventLoop")) {
                     this.geyser.getLogger().debug("EventLoop type is NIO because native event loops are disabled.");
                 } else {
-                    this.geyser.getLogger().debug("Reason for no Epoll: " + Epoll.unavailabilityCause().toString());
-                    this.geyser.getLogger().debug("Reason for no KQueue: " + KQueue.unavailabilityCause().toString());
+                    // Use lambda here, not method reference, or else NoClassDefFoundError for Epoll/KQueue will not be caught
+                    this.geyser.getLogger().debug("Reason for no Epoll: " + throwableOrCaught(() -> Epoll.unavailabilityCause()));
```

Comment:
Might want to also gate these statements in a debug mode check just to speed up processing?

Reply:
It already is, at the very top